### DISCUSSION TITLE
Add file name also on schema_migrations

### DISCUSSION THREAD

**Discussion by ap-in-git** [amacneil/dbmate#484] [Feature Ideas]
I think it would be better if we can add a file name also on schema_migrations other than versions. It makes things things easier to understand when you see schema_migrations to know what migrations has been applied.
For e.g. Laravel migrations schema keeps the file name also on migrations table.

If this features sounds okay. Maybe I can start working on PR for the pull request?

**Comment by amacneil** [comment]
I think that's ok, as long as we still only compare the version number (i.e. if you rename a file, but keep the same version number, it still counts as applied).

I would probably use the migration name rather than the full file name.

Also, there will need to be smooth and backwards-compatible upgrade path.

**Comment by tombohub** [comment]
What's the status of this?

**Reply by amacneil** [reply]
No one is currently working on it, and I'm skeptical of the value it brings, it feels against the dbmate minimalist philosophy.

For example someone is going to look at the schema_migrations table and if they see the file name they will start wondering how the file name affects migrations. Currently it is pretty clear that only the schema version number matters.
